### PR TITLE
Allow setting custom-entries-file in formats

### PR DESCRIPTION
**Pull request by liornoy**
This commit adds the the flag customEntriesFormat that sets the custom entries file format (json/yaml/csv).

We also include 3 example files to help the user to create his own custom file, and they'll be used in the integration test.

### CODE REVIEW

File: commatrix/commatrix.go [hunk 1]
```diff
@@ -34,7 +37,7 @@ const (
 // Custom entries from a JSON file can be added to the matrix by setting `customEntriesPath`.
 // Returns a pointer to ComMatrix and error. Entries include traffic direction, protocol,
 // port number, namespace, service name, pod, container, node role, and flow optionality for OpenShift.
-func New(kubeconfigPath string, customEntriesPath string, e Env, d Deployment) (*types.ComMatrix, error) {
+func New(kubeconfigPath string, customEntriesPath string, customEntriesFormat string, e Env, d Deployment) (*types.ComMatrix, error) {
```
Comment (sabinaaledort) [inline commatrix/commatrix.go:40]: customEntriesFormat should be an enum

File: commatrix/commatrix.go [hunk 2]
```diff
@@ -61,7 +64,10 @@ func New(kubeconfigPath string, customEntriesPath string, e Env, d Deployment) (
 	res = append(res, staticEntries...)
 
 	if customEntriesPath != "" {
-		customComDetails, err := addFromFile(customEntriesPath)
+		if customEntriesFormat == "" {
```
Comment (sabinaaledort) [inline commatrix/commatrix.go:67]: Should be validated in main

File: types/types.go [hunk 3]
```diff
@@ -51,7 +51,7 @@ func ToCSV(m ComMatrix) ([]byte, error) {
 }
 
 func ToJSON(m ComMatrix) ([]byte, error) {
-	out, err := json.Marshal(m.Matrix)
+	out, err := json.MarshalIndent(m.Matrix, "", "    ")
```
Comment (sabinaaledort) [inline types/types.go:54]: Why this change?

File: commatrix/commatrix.go (hunk 1 above)
Comment (liornoy) [reply commatrix/commatrix.go:40]: done, the flag still accepts string, but internally it's validated as an enum

File: commatrix/commatrix.go (hunk 2 above)
Comment (liornoy) [reply commatrix/commatrix.go]: done
